src/Type/User/Mutation/UserCreate.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace WPGraphQL\Type\User\Mutation;
+
+use GraphQLRelay\Relay;
+use WPGraphQL\Types;
+
+/**
+ * Class UserCreate
+ *
+ * @package WPGraphQL\Type\User\Mutation
+ */
+class UserCreate {
+
+	/**
+	 * Stores the user create mutation
+	 *
+	 * @var array $mutation
+	 * @access private
+	 */
+	private static $mutation;
+
+	/**
+	 * Process the user creat mutation
+	 *
+	 * @return array|null
+	 * @access public
+	 */
+	public static function mutate() {
+
+		if ( empty( self::$mutation ) ) {
+
+			self::$mutation = Relay::mutationWithClientMutationId( [
+				'name' => 'createUser',
+				'description' => __( 'Create new user object', 'wp-graphql' ),
+				'inputFields' => self::input_fields(),
+				'outputFields' => [
+					'user' => [
+						'type' => Types::user(),
+						'resolve' => function( $payload ) {
+							return get_user_by( 'ID', $payload['id'] );
+						}
+					]
+				],
+				'mutateAndGetPayload' => function( $input ) {
+
+					if ( empty( $input ) || ! is_array( $input ) ) {
+						throw new \Exception( __( 'Mutation not processed. There was no input for the mutation.', 'wp-graphql' ) );
+					}
+
+					if ( ! current_user_can( 'create_users' ) ) {
+						throw new \Exception( __( 'Sorry, you are not allowed to create a new user.', 'wp-graphql' ) );
+					}
+
+					/**
+					 * Map all of the args from GQL to WP friendly
+					 */
+					$user_args = UserMutation::prepare_user_object( $input, 'userCreate' );
+
+					/**
+					 * Create the new user
+					 */
+					$user_id = wp_insert_user( $user_args );
+
+					/**
+					 * Throw an exception if the post failed to create
+					 */
+					if ( is_wp_error( $user_id ) ) {
+						$error_message = $user_id->get_error_message();
+						if ( ! empty( $error_message ) ) {
+							throw new \Exception( esc_html( $error_message ) );
+						} else {
+							throw new \Exception( __( 'The object failed to create but no error was provided', 'wp-graphql' ) );
+						}
+					}
+
+					/**
+					 * If the $post_id is empty, we should throw an exception
+					 */
+					if ( empty( $user_id ) ) {
+						throw new \Exception( __( 'The object failed to create', 'wp-graphql' ) );
+					}
+
+					/**
+					 * Return the new user ID
+					 */
+					return [
+						'id' => $user_id,
+					];
+
+				}
+
+			] );
+		}
+
+		return ( ! empty( self::$mutation ) ) ? self::$mutation : null;
+
+	}
+
+	/**
+	 * Add the email as a nonNull field for update mutations
+	 *
+	 * @return array
+	 */
+	private static function input_fields() {
+
+		$user_fields = UserMutation::input_fields();
+
+		/**
+		 * Create mutations require a login to be passed
+		 */
+		$user_fields['login']['type'] = Types::non_null( Types::string() );
```

Comment:
since the `UserMutation::input_fields()` is filterable we should probably explicitly make sure the login field exists, similar to how you're doing here: https://github.com/wp-graphql/wp-graphql/pull/191/files#diff-4fdbc70dee8938bb749e25fefa47a18bR114

Let's say someone did a heavy-handed filter like: 

```
add_filter( 'graphql_user_mutation_input_fields', function() {
  return $fields['the_only_mutation_field'] = [
     'type' => Types::string(),
     'description' => __( 'I just filtered out all mutation inputs except this field, bro', 'some-silly-plugin' ),
  ];
}, 10, 1 );
```

This would cause: `$user_fields['login']['type'] = Types::non_null( Types::string() );` to be a PHP warning as the array_key `login` wouldn't exist.